### ISSUE TITLE
[BUG] Glob matching is broken in 7.0.12

### ISSUE THREAD

**Issue by CliffS** [npm/cli#2193]
### Current Behavior:

I have a script that creates a checksum for my coffee files:
```json
"scripts": {
    "checksum": "cat *coffee | cksum | cut -d' ' -f1"
}
```
Since 7.0.12 this gives:
`cat: '*coffee': No such file or directory`

### Expected Behavior:

In version 7.0.11 it correctly returns:
`4069467209`

### Steps To Reproduce:

In Linux attempt to use a shell glob in a script.  It fails.

### Environment:

OS: Ubuntu 20.04.10 LTS
Node: v15.2.1
npm: 7.0.12

**Comment by nlf** [comment]
i'm unable to reproduce this, i get the same error in every version i tested (7.0.8, 7.0.11, 7.0.12, 7.0.13, and even 6.14.9 and 6.14.8)

```
❯ npm run checksum

> temp@1.0.0 checksum
> cat *coffee | cksum | cut -d' ' -f1

cat: *coffee: No such file or directory
4294967295
```

however, I noticed something, if i run

```
> cat /dev/null | cksum | cut -d' ' -f1
4294967295
```

I get the exact same output I get when trying the npm script, this is because `cat` is closing stdin and `cksum` is calculating the checksum of nothing. Is it possible something on your end was hiding stderr output previously and no longer is?

**Comment by CliffS** [comment]
> i'm unable to reproduce this, i get the same error in every version i tested (7.0.8, 7.0.11, 7.0.12, 7.0.13, and even 6.14.9 and 6.14.8)

Well you would, unless you have coffee files in the current path.  You are correct that 4294967295 is the checksum of null.

However, I think I may have been unclear in the original bug report.  There **are** coffee files in the path.  The issue is that `*.coffee` is not expanding them, it is being treated as a literal.

If this makes it any clearer, at a shell prompt, if you type `ls -l *` you should get a listing of al the non-hidden files in the directory.  If, on the other hand, you typed `ls -l '*'` you would get `ls: cannot access '*': No such file or directory` because the `*` is now a literal.

This is what is happening inside npm.  in my case, `*coffee` should be expanded to a list of coffee files in the directory but it is being treated as if it had quotes around it.  You may notice that, in my original example under v7.0.11, the resultant checksum was `4069467209` rather than the null value of `4294967295`.

Is that any clearer?

**Comment by nlf** [comment]
you were absolutely right! my mistake! the pull request referenced above will resolve the issue and be part of our next release

**Comment by CliffS** [comment]
> the pull request referenced above will resolve the issue

Wonderful.  Thank you so much for dealing with this so promptly.

**Comment by nlf** [comment]
published 7.0.14 that fixes this. thanks again for the report!
